src/components/icons/helpers/withIconSvg.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react';
+import classnames from 'classnames';
+import * as styles from './withIconSvg.scss';
+
+/**
+ * Note: need to export to avoid error in Icons namespace.
+ */
+export interface IconSvgProps {
+	className?: string;
+	key?: string | number;
+	id?: string;
+	style?: object;
+}
+
+export interface IconSvgStoryMetaAdditionalProps {
+	/* the name of the icon prop */
+	propName: string;
+	/* the name of the component type used to allow changes to the prop */
+	type: 'select';
+	/* the options for the component type */
+	options: any[];
+	/* the default option */
+	default: any;
+}
+
+/**
+ * Typing for icon meta data used for Storybook.
+ */
+export interface IconSvgStoryMeta {
+	/* manually included props used to disply in storybook and allow the user to change options */
+	additionalProps?: IconSvgStoryMetaAdditionalProps[];
+	/* additional search terms as used by the storybook example */
+	tags: string[];
+}
+
+/**
+ * HOC that adds additional props and styles to icons.
+ * @param IconComponent The icon component to be decorated by the hoc.
+ * @param neutralFill Whether to set the color of the svg path fill to the theme's neutral gray color.
+ */
+const _withIconSvg = <P extends object>(
+	IconComponent: any,
+	neutralFill: boolean,
+): React.FC<P & IconSvgProps> => ({
+	...props
+}) => {
+	const {
+		className,
+		id,
+		key,
+		style,
+		...otherProps
+	} = props;
+
+	const render = (
+		<IconComponent
+			className={classnames(
+				styles.IconSvg,
+				'IconSvg',
+				className,
+				{
+					[styles.IconSvg__neutralizeFill]: neutralFill,
+				},
+			)}
+			id={id}
+			key={key}
+			style={style}
+			{...otherProps as P}
+		/>
+	);
+
+	return render;
+}
+
+/**
+ * HOC that adds additional props and styles to icons.
+ * @param IconComponent The icon component to be decorated by the hoc.
+ * @param neutralFill Whether to set the color of the svg path fill to the theme's neutral gray color.
+ * @param meta The meta details for an svg that makes an icon more discoverable within storybook example.
+ */
+const withIconSvg = <P extends object>(
+	IconComponent: any,
+	neutralFill: boolean,
```

Comment:
It looks like the majority of the calls to this function are getting `true` as their second arg. Thoughts on flipping the order of `neutralFill` and `meta` and then making the default value of `neutralFill` `true`?

Something like

```js
const withIconSvg = <P extends object>(
	IconComponent: any,
	meta: IconSvgStoryMeta,
	neutralFill: boolean = true,
): React.FC<P & IconSvgProps> => {
// ...
}
```

Reply:
It's a great thought. I actually did this initially to make a developer understand how they're implementing every given SVG and deciding whether a neutral fill actually makes sense. My goal is to work with Tyson and figure out a better way to do this and ultimately introduce a color prop for all icon components that makes this go away and automatically neutralizes all of them. Or, even better yet, let's get already neutralized assets and get rid of the ugly business of doing that behind-the-scenes. Let me talk to Tyson and figure out what we want to do here.

Reply:
I talked with Tyson and now have a plan to bake these values into the svg and that'll come in a soon-to-follow release. Thanks for the feedback!